### PR TITLE
Fixed issue where Session authorization state overrode token

### PR DESCRIPTION
**Pull request by thongsav-usgs**
authorization state. Token state is always checked now.

### CODE REVIEW

File: auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityContextFilter.java [hunk 1]
```diff
@@ -52,23 +52,17 @@
...
-    	}
-    	
-        LOG.trace("Session authorized: {}", sessionAuthorized);
-        LOG.trace("Token authorized: {}", tokenAuthorized);
-    }
+	@Override
+	public void filter(ContainerRequestContext requestContext) {
+		boolean tokenAuthorized = false;
+		try {
+			if(SecurityContextUtils.isTokenAuthorized(requestContext, httpRequest, getAuthClient(), getAdditionalRoles())) { //will have to populate the security context
```
Comment (thongsav-usgs) [inline auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityContextFilter.java]: This is the only line that really changed here, everything else is white space fixes.

File: auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityFilter.java [hunk 2]
```diff
@@ -60,24 +60,23 @@
...
-	            .entity("User cannot access the resource.")
-	            .build());
-    }
+	@Override
+	public void filter(ContainerRequestContext requestContext) throws IOException {
+		//if root documentation or authenticate url, continue on without authorization
+		String requestedUri = requestContext.getUriInfo().getPath();
+		List<String> allowedUris = getUnsecuredUris();
+		if(allowedUris == null || !allowedUris.contains(requestedUri)) {
+			if (!SecurityContextUtils.isTokenAuthorized(requestContext, httpRequest, getAuthClient(), getAdditionalRoles())) {
```
Comment (thongsav-usgs) [inline auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityFilter.java]: Same, this line changed, everything else is white space.

File: auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityContextFilter.java (hunk 1 above)
Comment (jiwalker-usgs) [inline auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityContextFilter.java]: Is anything lost by removing the session stuff that was removed?

Comment (thongsav-usgs) [inline auth-web-utils/src/main/java/gov/usgs/cida/auth/ws/rs/filter/AbstractTokenBasedSecurityContextFilter.java]: Not really, it still does mostly the same thing under the hood (Line 75 of HttpTokenUtils.java). The only thing "lost" is it no longer shortcuts validating the token, which is probably the best thing to do anyway. It gets more chatty with the backend auth service if you aren't using the caching client.

**Comment by jiwalker-usgs** [comment]
:shipit:

**Pull request merged**